priv/static/phoenix_live_view.js

```diff
@@ -2104,6 +2109,87 @@ removing illegal node: "${(childNode.outerHTML || childNode.nodeValue).trim()}"
   };
 
   // js/phoenix_live_view/rendered.js
+  var VOID_TAGS = new Set([
+    "area",
+    "base",
+    "br",
+    "col",
+    "command",
+    "embed",
+    "hr",
+    "img",
+    "input",
+    "keygen",
+    "link",
+    "meta",
+    "param",
+    "source",
+    "track",
+    "wbr"
+  ]);
+  var endingTagNameChars = new Set([">", " ", "\n", "	", "\r"]);
+  var modifyRoot = (html, attrs, clearInnerHTML) => {
+    let i = 0;
+    let insideComment = false;
+    let insideTag = false;
+    let tag;
+    let beforeTagBuff = [];
+    while (i < html.length) {
+      let char = html.charAt(i);
+      if (insideComment) {
+        if (char === "-" && html.slice(i, i + 3) === "-->") {
+          insideComment = false;
+          beforeTagBuff.push("-->");
+          i += 3;
+        } else {
+          beforeTagBuff.push(char);
+          i++;
+        }
+      } else if (char === "<" && html.slice(i, i + 4) === "<!--") {
+        insideComment = true;
+        beforeTagBuff.push("<!--");
+        i += 4;
+      } else if (char === "<") {
+        insideTag = true;
+        let iAtOpen = i;
+        for (i; i < html.length; i++) {
+          if (endingTagNameChars.has(html.charAt(i))) {
+            break;
+          }
+        }
+        tag = html.slice(iAtOpen + 1, i);
+        break;
+      } else if (!insideComment && !insideTag) {
+        beforeTagBuff.push(char);
```

Comment:
I am not even sure if this can help. Or rather, if we are not inside the comment, we could just ignore it, but it is fine to keep it around too. If so, I would keep it as a else:

```suggestion
      } else {
        beforeTagBuff.push(char);
```